src/ocean/task/TaskPool.d

```diff
@@ -153,6 +153,36 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
         return true;
     }
 
+    /***************************************************************************
+
+        Suspends the current task until all tasks in the pool have finished
+        executing.
+
+    ***************************************************************************/
+
+    public void awaitAllTasks ()
+    {
+        scope iter = this.new ReadOnlyBusyItemsIterator;
```

Comment:
Yep you're right. I overlooked the fact that I'm modifying the pool item when I add the termination hook. I should be using `BusyItemsIterator` instead.